### PR TITLE
v4 Travis Build

### PR DESCRIPTION
**Pull request by thomas-mcdonald**
probably a tonne of cleaning out to do here, here's a branch to work on it.

### CODE REVIEW

**Comment by kkirsche** [comment]
Duplicate of #17012

**Comment by thomas-mcdonald** [comment]
no it's not

**Comment by kkirsche** [comment]
Sorry, dupe just a different branch.

**Comment by thomas-mcdonald** [comment]
@cvrebert Seems to be mostly there wrt. building - scss-lint is failing in a big way & sauce labs is timing out though...

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 language: node_js
 git:
   depth: 10
 node_js:
   - "0.12"
 before_install:
-  - travis_retry sudo pip install -r test-infra/requirements.txt
-  - rvm use 1.9.3 --fuzzy
+  - export PATH=$HOME/.local/bin:$PATH
+  - travis_retry pip install -r test-infra/requirements.txt --user `whoami`
```
Comment (cvrebert) [inline .travis.yml]: `--user` doesn't take an argument, so the `whoami` should be unnecessary

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,13 @@
+sudo: false
 language: node_js
 git:
   depth: 10
 node_js:
   - "0.12"
 before_install:
-  - travis_retry sudo pip install -r test-infra/requirements.txt
-  - rvm use 1.9.3 --fuzzy
+  - export PATH=$HOME/.local/bin:$PATH
```
Comment (cvrebert) [inline .travis.yml]: We only use the libraries that pip installs, not the executables, so this should be unnecessary

File: .travis.yml [hunk 3]
```diff
@@ -13,12 +15,19 @@ before_install:
   - echo "$TRAVIS_COMMIT_MSG" | grep '\[skip sauce\]'; export TWBS_DO_SAUCE=$?; true
   - if [ "$TRAVIS_REPO_SLUG" = twbs-savage/bootstrap ]; then export TWBS_DO_VALIDATOR=0; fi
 install:
+  - bundle install --deployment --jobs=3
+  - gem install scss_lint --no-rdoc --no-ri
```
Comment (cvrebert) [inline .travis.yml]: We have a Gemfile.lock for this... https://github.com/twbs/bootstrap/blob/v4-dev/test-infra/gemfiles/core.gemfile.lock

Comment (thomas-mcdonald) [inline .travis.yml]: there's one in root which bundle install is using, but --deployment vendors them locally for caching & whichever grunt scss lint plugin we use doesn't support bundle exec

Comment (cvrebert) [inline .travis.yml]: Perhaps we should take out the middleman and axe that grunt plugin

**Comment by thomas-mcdonald** [comment]
@cvrebert could you take a look at the Sauce Labs errors? I think this is hopefully almost ready to go.

**Comment by cvrebert** [comment]
The Sauce failures are due to #17097.

**Pull request merged**